### PR TITLE
NAS-132648 / 25.04 / query param handling for auth tokens

### PR DESCRIPTION
**Pull request by aervin**
**Changes:**

**Testing:**
One method:
* clear `localStorage`
* you'll have to comment out the `https:` check in `AuthService.setQueryToken`
* `midclt call auth.generate_token`
* append the token to an app route: `localhost:4200/storage/disks?token=<token>`

### Downstream

|Affects         |Reasoning
|----------------|-------------------------------
|Documentation   |

### CODE REVIEW

**Comment by aervin** [comment]
The approach so far is to forward query params on redirect to `/signin` (`AuthGuardService:canActivate`). From there, add a bit of logic to `signin.store` and `auth.service` to handle the token.

However, this isn't working like I'd expect. Seems that `WebsocketHandlerService` is calling its `resetUI` method and interfering with the redirect.

image

**Review by undsoft — COMMENTED** [review]
Seems to be redirecting fine for me: 

https://github.com/user-attachments/assets/c75045a4-d1b8-496f-b0d2-53d4a0cfce22

**Comment by undsoft** [comment]
BTW, I don't know if tokens in browser history is kosher.

File: src/app/pages/signin/store/signin.store.ts [hunk 1]
```diff
@@ -35,13 +35,15 @@ interface SigninState {
...
   loginBanner: string;
+  queryToken: string | null;
 }
 
 const initialState: SigninState = {
   isLoading: false,
   wasAdminSet: true,
   failover: null,
   loginBanner: null,
+  queryToken: null,
```
Comment (undsoft) [inline src/app/pages/signin/store/signin.store.ts:46]: I don't think it needs to be part of state here and have a selector. It's not used outside of the component.

File: src/app/pages/signin/store/signin.store.ts [hunk 2]
```diff
@@ -240,8 +248,14 @@ export class SigninStore extends ComponentStore<SigninState> {
   }
 
   private handleLoginWithToken(): Observable<LoginResult> {
```
Comment (undsoft) [inline src/app/pages/signin/store/signin.store.ts:250]: Method gets a bit convoluted with all the conditions. How about introducing a new method for logging in with query token and either calling `handleLoginWithToken` or that new method depending on whether query parameter set.

**Review by undsoft — APPROVED** [review]
The code looks good, but I just couldn't get it to work on a CI machine.
I can see token being sent to middleware, but it just returns `AUTH_ERR` for some reason.

**Review by RehanY147 — APPROVED** [review]

**Comment by aervin** [comment]
Went to retest against a CI machine and the `hasOpened` logic was already flaky/broken. As an alternative approach, I've manually preserved query params on redirect in the `resetUi` method.

Seems to work fine against a CI machine with both `root` and non-builtin users. To prove I'm not crazy:
nas-redirect

**Review by undsoft — COMMENTED** [review]
I still get the same issue:

https://github.com/user-attachments/assets/24c2d79c-44e6-43c4-a666-076923222e1b

**Pull request merged**